src/transformers/integrations/integration_utils.py

```diff
@@ -1455,6 +1456,10 @@ def __init__(self):
 class NeptuneCallback(TrainerCallback):
```

Comment:
I'm not very familiar with the modern `transformers` deprecation approach, but do we have a nice decorator here that can inform IDE's that objects are deprecated, perhaps?
<img width="472" height="159" alt="image" src="https://github.com/user-attachments/assets/4b14ffe0-9c6c-43d1-9e2b-c2fb32ac8025" />